datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -115,33 +115,37 @@ func (f *ChainDataFetcher) Start(server p2p.Server) error {
 
 	if !f.config.NoDefaultStart {
 		if err := f.startFetching(); err != nil {
+			logger.Error("start fetching is failed", "err", err)
 			return err
 		}
 	}
 
 	go f.resLoop()
-
+	logger.Info("chaindata fetcher is started", "numHandlers", f.numHandlers)
 	return nil
 }
 
 func (f *ChainDataFetcher) Stop() error {
 	f.stopFetching()
 	f.stopRangeFetching()
+	logger.Info("wait for all goroutines to be terminated...", "numGoroutines", f.config.NumHandlers)
 	close(f.stopCh)
-	logger.Info("wait for all goroutines to be terminated...")
 	f.wg.Wait()
-	logger.Info("terminated all goroutines for chaindatafetcher")
+	logger.Info("chaindata fetcher is stopped")
 	return nil
 }
 
-func (f *ChainDataFetcher) makeRequests(start, end uint64, reqType requestType, stopCh chan struct{}) {
-	for i := start; i <= end; i++ {
+func (f *ChainDataFetcher) makeRequests(startBlock, endBlock uint64, reqType requestType, stopCh chan struct{}) {
```

Comment:
```suggestion
func (f *ChainDataFetcher) sendRequests(startBlock, endBlock uint64, reqType requestType, stopCh chan struct{}) {
```

Reply:
I rebased the PR, and it is resolved. :)